scipy/integrate/_quadrature.py

```diff
@@ -312,11 +313,19 @@ def vfunc(x):
     return vfunc
 
 
+@_deprecated("`scipy.integrate.quadrature` is deprecated as of SciPy 1.12.0"
+             "and will be removed in SciPy 1.14.0. Please use"
+             "`scipy.integrate.quad` instead.")
 def quadrature(func, a, b, args=(), tol=1.49e-8, rtol=1.49e-8, maxiter=50,
                vec_func=True, miniter=1):
     """
     Compute a definite integral using fixed-tolerance Gaussian quadrature.
 
+    .. deprecated:: 1.12.0
```

Comment:
Minor nit but if we put the directive first in the docstring then the deprecation also shows up in the toc too like
![image](https://github.com/scipy/scipy/assets/60778417/77b1e183-803f-4435-8b17-122017310a8a)